python/grass/jupyter/timeseries.py

```diff
@@ -0,0 +1,435 @@
+
+        Adds overlays and legend to base map.
+        """
+        img = GrassRenderer(
+            filename=filename, use_region=True, env=self._env, read_file=True
+        )
+        for grass_module, kwargs in self._overlays.calls:
+            img.run(grass_module, **kwargs)
+        # Add legend if needed
+        if self._legend:
+            self._render_legend(img)
+
+    def render(self):
+        """Renders image for each time-step in space-time dataset.
+
+        Save PNGs to temporary directory. Must be run before creating a visualization
+        (i.e. time_slider or animate). Can be time-consuming to run with large
+        space-time datasets.
+        """
+
+        # Make base image (background and baselayers)
+        random_name_base = gs.append_random("base", 8) + ".png"
+        base_file = os.path.join(self._tmpdir.name, random_name_base)
+        img = GrassRenderer(
+            filename=base_file, use_region=True, env=self._env, read_file=True
+        )
+        # Fill image background
+        img.d_erase(bgcolor=self._bgcolor)
+        # Add baselayers
+        for grass_module, kwargs in self._baselayers.calls:
+            img.run(grass_module, **kwargs)
+
+        # Create name for empty layers
+        random_name_none = gs.append_random("none", 8) + ".png"
+
+        # Render each layer
+        for date, layer in self._date_layer_dict.items():
+            if layer == "None":
+                # Create file
+                filename = os.path.join(self._tmpdir.name, random_name_none)
+                self._date_filename_dict[date] = filename
+                # Render blank layer if it hasn't been done already
+                if not os.path.exists(filename):
+                    shutil.copyfile(base_file, filename)
+                    self._render_blank_layer(filename)
+            else:
+                # Create file
+                filename = os.path.join(self._tmpdir.name, f"{layer}.png")
+                shutil.copyfile(base_file, filename)
+                self._date_filename_dict[date] = filename
+                # Render image
+                img = GrassRenderer(
+                    filename=filename, use_region=True, env=self._env, read_file=True
+                )
+                if self._element_type == "strds":
+                    img.d_rast(map=layer)
+                elif self._element_type == "stvds":
+                    img.d_vect(map=layer)
+                for grass_module, kwargs in self._overlays.calls:
+                    img.run(grass_module, **kwargs)
+                # Add legend if needed
+                if self._legend:
+                    self._render_legend(img)
+        self._layers_rendered = True
+
+    def time_slider(self, slider_width=None):
+        """Create interactive timeline slider.
+
+        param str slider_width: width of datetime selection slider as a
+                                percentage (%) or pixels (px)
```

Comment:
Maybe just short description here and then describe the format in a separate paragraph. It needs some more doc because it is a string with a suffix and that's not clear.